### DISCUSSION TITLE
Multiple CRS support

### DISCUSSION THREAD

**Discussion by gchoqueux** [iTowns/itowns#1777] [Next Features]
### Feature description

- Add the **multiple crs projections support**.
- Visualize simultaneously vector or raster data with different projections

### How to develop it
There are different way:

1. The **projection texture** : use shader to re project data
2. Use projection grid mesh to re project data (resampling data with a vector geometry)

#### Map box
- [reprojection example](https://docs.mapbox.com/mapbox-gl-js/example/projections/)
- [reprojection information](https://www.mapbox.com/blog/adaptive-projections )
#### Openlayer
- [reprojection example](https://openlayers.org/en/latest/examples/reprojection.html)
- [reprojection information](https://openlayers.org/doc/tutorials/raster-reprojection.html)

**Comment by jailln** [comment]
Nice initiative! My inputs on this subject:

1. I think we should clarify what is the actual support of itowns on this subject. For what I know, it seems that:

-  It is already supported for EPSG:4978 and 3857 in globe view
- terrain elevation can be passed in any CRS and is reprojected in the view crs

Is that correct ? If yes, we should indicate it clearly in the documentation, either in the corresponding classes or maybe in a master documentation describing the general philosophy and features of itowns. Maybe something like a developer guide for users of the API like what's done for [deck.gl](https://deck.gl/docs/developer-guide/coordinate-systems)?

2. Could you elaborate on the two options you provide ? And what are the pros and cons ? Is the first way for rasters and the second for vectors ?

**Reply by gchoqueux** [reply]
> I think we should clarify what is the actual support of itowns on this subject. For what I know, it seems that:

The globe view support 
- For `ColorLayer` : `EPSG:3857` and `EPSG:4326`
- For `ElevationLayer`  :  `EPSG:4326`
- For `GeometryLayer` with vector source  :  Any projections

We will add clarifications in [itowns tutorials](http://www.itowns-project.org/itowns/docs/#home).

> Is the first way for rasters and the second for vectors ?

It's two way to display data on terrain with `ColorLayer` or `ElevationLayer`.
I think the best way is projection grid mesh, this's the choice of OpenLayer and MapBox.
I will detail later the technique and the developments
